### PR TITLE
Updates to tests for simpler external ID API.

### PR DESCRIPTION
**Pull request by alxdarksage**

### CODE REVIEW

**Review by DwayneJengSage — APPROVED** [review]
Looks good. Just some minor comments about comments, and one question.

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ExternalIdsV4Test.java [hunk 1]
```diff
@@ -69,172 +74,126 @@ public void test() throws Exception {
...
-            // allow the assignment of multiple external IDs on sign up. Adding new studies is probably
-            // going to happen by signing additional consents, but that's TBD.
-            SignUp signUp = new SignUp().appId(TEST_APP_ID);
-            signUp.setPassword(Tests.PASSWORD);
-            signUp.setExternalId(extIdA);
-            researcher.getClient(AuthenticationApi.class).signUp(signUp).execute();
+            userId1 = researcherApi.createParticipant(new SignUp().externalIds(ImmutableMap.of(STUDY_ID_1, extIdA))).execute().body().getIdentifier();
+            usersToDelete.add(userId1);
+            // researcherApi.createParticipant(new SignUp().externalIds(ImmutableMap.of(STUDY_ID_2, extIdB1))).execute().body().getIdentifier();
+            // researcherApi.createParticipant(new SignUp().externalIds(ImmutableMap.of(STUDY_ID_2, extIdB2))).execute().body().getIdentifier();
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ExternalIdsV4Test.java:87]: removed

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/ParticipantsTest.java [hunk 2]
```diff
@@ -91,15 +90,15 @@
...
+    private String externalId;
     
     @Before
     public void before() throws Exception {
         admin = TestUserHelper.getSignedInAdmin();
         developer = TestUserHelper.createAndSignInUser(ParticipantsTest.class, false, DEVELOPER);
         researcher = TestUserHelper.createAndSignInUser(ParticipantsTest.class, true, RESEARCHER);
         
-        externalId = Tests.createExternalId(ParticipantsTest.class, developer, STUDY_ID_1);
+        externalId = Tests.randomIdentifier(ParticipantsTest.class); // Tests.createExternalId(ParticipantsTest.class, developer, STUDY_ID_1);
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/ParticipantsTest.java:101]: removed

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SignInTest.java [hunk 3]
```diff
@@ -90,7 +93,7 @@ public void canGetDataGroups() throws Exception {
     public void createComplexUser() throws Exception {
         AuthenticationApi authApi = researcher.getClient(AuthenticationApi.class);
         
-        ExternalIdentifier externalId = Tests.createExternalId(SignInTest.class, developer, STUDY_ID_1);
+        String externalId = Tests.randomIdentifier(SignInTest.class);//, developer, STUDY_ID_1);
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/SignInTest.java:96]: removed

File: src/test/java/org/sagebionetworks/bridge/sdk/integration/SignInTest.java [hunk 4]
```diff
@@ -103,37 +106,43 @@ public void createComplexUser() throws Exception {
...
         signUp.setAttributes(map);
-                
-        authApi.signUp(signUp).execute();
 
+        authApi.signUp(signUp).execute();
+        
+        TestUser admin = TestUserHelper.getSignedInAdmin();
+        
+        // User can no longer enroll themself in a study, but until the user is enrolled, the researcher
+        // cannot see the account. Enrollment the participant.
```
[the remark this answers is not in the document]
Comment (alxdarksage) [reply src/test/java/org/sagebionetworks/bridge/sdk/integration/SignInTest.java:121]: We don't have an answer yet for what study people enroll in if they download an app and it has multiple studies...for MTB people are being enrolled with external IDs in specific studies. it's possible one study will be marked as the default public study that people have to consent to first. Once people are enrolled in a study, a researcher who can see the study can see the enrolled participants.

**Review by DwayneJengSage — APPROVED** [review]

**Pull request merged**